lib/fileio/primUpdater.h

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright 2018 Pixar
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef PXRUSDMAYA_MAYAPRIMUPDATER_H
+#define PXRUSDMAYA_MAYAPRIMUPDATER_H
+
+/// \file usdMaya/primUpdater.h
+
+#include "../base/api.h"
+#include "primUpdaterContext.h"
+#include "../utils/util.h"
+
+#include "pxr/pxr.h"
+#include "pxr/usd/sdf/path.h"
+
+#include <maya/MDagPath.h>
+#include <maya/MFnDependencyNode.h>
+#include <maya/MObject.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+class UsdMayaPrimUpdater
+{
+public:
+    MAYAUSD_CORE_PUBLIC
+    UsdMayaPrimUpdater(const MFnDependencyNode& depNodeFn,
+                       const SdfPath& usdPath);
+    
+    UsdMayaPrimUpdater() = default;
+    virtual ~UsdMayaPrimUpdater() = default;
+
+    MAYAUSD_CORE_PUBLIC
```

Comment:
This export macro should be removed. Our build fails because of `-Werror=attributes`:
```
maya_open_source/maya-usd/lib/fileio/primUpdater.h:45:16: error: attribute ignored in declaration of 'enum class UsdMayaPrimUpdater::Supports' [-Werror=attributes]
```